include/pybind11/cast.h

```diff
@@ -2333,39 +2378,20 @@ class unpacking_collector {
     }
 
 private:
-    tuple m_args;
-    dict m_kwargs;
+    small_vector<object, arg_vector_small_size> m_args;
+    object m_names; // null or a tuple of names
```

Comment:
```suggestion
    tuple m_names; // null or a tuple of names
```
Hey to break it to you but all py::object can be a null including py::tuple (it's the default moved from state). Also reduces the need for explict reintrpret_borrow and casting everywhere.